plugins/woocommerce/includes/class-wc-shipping.php

```diff
@@ -358,6 +358,27 @@ public function calculate_shipping_for_package( $package = array(), $package_key
 				}
 			}
 
+			// Hide shipping rates when free shipping is available.
+			if ( 'yes' === get_option( 'woocommerce_shipping_hide_rates_when_free', 'no' ) ) {
+				$free_shipping = array();
+				$local_pickup  = array();
+
+				foreach ( $package['rates'] as $rate ) {
+					if ( 'free_shipping' === $rate->method_id ) {
+						$free_shipping[] = $rate;
+						continue;
+					}
+
+					if ( in_array( $rate->method_id, array( 'local_pickup', 'pickup_location' ), true ) ) {
```

Comment:
There are other locations in the codebase where the list of pickup methods is filterable. `woocommerce_local_pickup_methods` Should that be used here? @ismaeldcom

Reply:
Also not just that, any custom shipping method that [declares itself as `pickup` compatible](https://github.com/woocommerce/woocommerce/blob/40e8ca877af1c7f3ef371102a07793a478330072/plugins/woocommerce/src/Blocks/Shipping/PickupLocation.php#L43) will be filtred out here, should ideally check the method `supports` for `'local-pickup'`, this should pickup `pickup_location` by default and any third party one. At least, we use `supports` for UI purposes as well.

Reply:
Thanks for the extra context!

I think applying the `woocommerce_local_pickup_methods` filter here makes sense. As we're filtering the rates, checking for the shipping method `supports` will add an extra overhead, which is already handled in the previous code block. And it will not cover the legacy one.

WDYT?

Reply:
I don't think there's much overhead as the method is already loaded in memory at that point, otherwise, you wouldn't get the shipping rates. I think it can be combined with `array_includes( $method->supports, 'local-pickup' ) || 'local_pickup' === $method->id`

Reply:
Follow-up in #56357

Reply:
Resolved in #56357. Thanks for your support @senadir